pkg/migrations/migrations_test.go

```diff
@@ -502,3 +505,217 @@ func TestCreateSystemTable(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestUpdateViewDependenciesMigration(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	ctx := context.Background()
+
+	// Start up a test server without running the view update migration. We also
+	// hijack the migration process to capture the SQL memory metric object, needed below.
+	var memMetrics *sql.MemoryMetrics
+	backwardCompatibleMigrations = []migrationDescriptor{
+		{
+			name:   "create system.settings table",
+			workFn: createSettingsTable,
+		},
+		{
+			name: "capture mem metrics",
+			workFn: func(ctx context.Context, r runner) error {
+				memMetrics = r.memMetrics
+				return nil
+			},
+		},
+	}
+	var s serverutils.TestServerInterface
+	var kvDB *client.DB
+
+	t.Run("starting server", func(t *testing.T) {
+		s, _, kvDB = serverutils.StartServer(t, base.TestServerArgs{})
+	})
+
+	defer s.Stopper().Stop(ctx)
+	e := s.Executor().(*sql.Executor)
+
+	t.Run("create test tables", func(t *testing.T) {
+		session := sql.NewSession(ctx, sql.SessionArgs{User: security.RootUser}, e, nil, memMetrics)
+		session.StartUnlimitedMonitor()
+		defer session.Finish(e)
+
+		const createStmts = `
+CREATE DATABASE test;
+CREATE DATABASE test2;
+SET DATABASE=test;
+
+CREATE TABLE t(x INT, y INT);
+CREATE VIEW v1 AS SELECT x FROM t WHERE false;
+CREATE VIEW v2 AS SELECT x FROM v1;
+
+CREATE TABLE u(x INT, y INT);
+CREATE VIEW v3 AS SELECT x FROM (SELECT x, y FROM u);
+
+CREATE VIEW v4 AS SELECT id from system.descriptor;
+
+CREATE TABLE w(x INT);
+CREATE VIEW test2.v5 AS SELECT x FROM w;
+`
+		res := e.ExecuteStatements(session, createStmts, nil)
+		defer res.Close(ctx)
+		if err := checkQueryResults(res.ResultList, 11); err != nil {
+			t.Fatal(err)
+		}
+	})
+
+	testDesc := []struct {
+		dbName parser.Name
+		tname  parser.Name
+		desc   *sqlbase.TableDescriptor
+	}{
+		{"test", "t", nil},
+		{"test", "v1", nil},
+		{"test", "u", nil},
+		{"test", "w", nil},
+		{"system", "descriptor", nil},
+	}
+
+	t.Run("fetch descriptors", func(t *testing.T) {
+		vt := e.GetVirtualTabler()
+
+		if err := kvDB.Txn(ctx, func(ctx context.Context, txn *client.Txn) error {
+			for i := range testDesc {
+				desc, err := sql.MustGetTableOrViewDesc(ctx, txn, vt, &parser.TableName{DatabaseName: testDesc[i].dbName, TableName: testDesc[i].tname}, true)
+				if err != nil {
+					return err
+				}
+				testDesc[i].desc = desc
+			}
+			return nil
+		}); err != nil {
+			t.Fatal(err)
+		}
+	})
+
+	// Now, corrupt the descriptors by breaking their dependency information.
+	t.Run("break descriptors", func(t *testing.T) {
```

Comment:
E.g., this will segfault if someone runs `make test PKG=./pkg/migration TESTS=TestUpdateViewDependenciesMigration/break_descriptors`. I feel like you might be better off with `t.Logf("break descriptors")`.